### ISSUE TITLE
vCenter!!!!!!

### ISSUE THREAD

**Issue by willnx** [willnx/vlab#16]
For whatever god damn reason, the _configure OneFS_ automation is broke after doing the MR upgrade to vCenter... I'm betting it's related to the fancy new login page :(

**Comment by willnx** [comment]
It's still broke. 

Not sure what's going on, but looks like the vCenter canvas which implements the HTML console has changed. Will update as I figure things out.

**Comment by willnx** [comment]
Holy shit, VMware really broke the HTML console with that 6.7.0.3 MR...

I've been able to work around the `canvas` no longer accepting [send_keys](https://selenium-python.readthedocs.io/api.html#selenium.webdriver.remote.webelement.WebElement.send_keys) by using an [ActionChain](https://selenium-python.readthedocs.io/api.html#module-selenium.webdriver.common.action_chains), but the `reset_actions` method doesn't seem to work. The result is that when I 1st click "yes" to format the disk, then later press "1" to create a new cluster, the original "yes" is also getting sent... 😒 ActionChains have a [pause](https://selenium-python.readthedocs.io/api.html#selenium.webdriver.common.action_chains.ActionChains.pause) method, so it might be possible to build up a _nightmare chain_ that would step through the config wizard, but this will require a significant amount of rework of the _auto config_ logic unless there's some sort of way to build up the ActionChain, and press the _"go button"_ at the very end.

Long story short; this is going to be broke until sometime next week. 😞

**Comment by willnx** [comment]
**Update**

The ActionChain API does enable me to build up the series of _things to do_ without a significant amount of re-work of the OneFS autoconfig logic. The biggest downside to this change will be that logging of the automation action cannot coincide with literal action. Like right now, when the logs say _"setting file system encoding to UTF-8"_ and you're watching the HTML5 console, you'll see the automation do what the logs said. With ActionChains this isn't possible, because you build up _all the things_ then hit a "go" button.

Also, I noticed that ActionChain has the repeating character issue I noticed when manually using the HTML5 console. Looks like it's specific to Firefox per VMware: https://kb.vmware.com/s/article/53103. So I'm going to have to update the service to use Chrome instead of Firefox.

On Monday, I'm going to make the switch to Chrome, and run some tests. Maybe the `send_keys` API works in Chrome, and I won't have to switch to ActionChains, because I really like the _real time logging_ of what the automation is doing.

**Comment by willnx** [comment]
Muhahahahaha! I decided to try Chrome _as is_ (i.e. without the ActionChains) once I got it working, and it's working!!!!!

I have some unit test to fix, then I'll have to build a pile of Docker containers, but by EoD today, this issue should be resolved! 😃

**Comment by RKasaraneni-7** [comment]
Hello Nick, 
I've created a Win-10 server using the below command - 
PS C:\Users\kasarr1> vlab create windows -i 10 -n spinach
Creating a new instance of Windows 10
Creating an RDP port mapping rule
Type     :  Windows
Version  :  10
State    :  poweredOn
IPs      :  fe80::c89b:e1d5:dd46:bc5c 192.168.1.164
Console  :  https://vlab.emc.com/api/1/link/706f9c28e6759951d2cb17629551dcf0
Use 'vlab connect windows --name spinach' to access your new Windows client

When I use -
PS C:\Users\kasarr1> vlab connect windows --name spinach
Lookin up connection information for spinach

it is starting the RDP session but after few minutes, it is throwing up an error. When I see the vlab status - 
PS C:\Users\kasarr1> vlab status
Collecting information about your inventory
Processing inventory records

Username: kasarr1
Gateway : 10.241.80.141

Machines:

 Name       | IPs           | Connectable   | Type    | Version   | Powered   | Console
------------+---------------+---------------+---------+-----------+-----------+--------------------------------
spinach    | 192.168.1.164 | False         | Windows | 10        | On        | https://vlab.emc.com/api/1/link/9079af9e714c985a87990415ccfca557

Connectable shows False for this whereas for my other instances, they are True. Can you please provide insight on this one. 
<img width="279" alt="Capture" src="https://user-images.githubusercontent.com/49874838/56681508-70330f00-6697-11e9-9055-a2b89f1c4e5e.PNG">

**Comment by bharpe** [comment]
Hi, I have been trying to create a solid lab setup to work with for customer recreates.  I am puzzled as to why half of my machines show connectable-true and the other half are connectable=false.

C:\Users\harpew>vlab status
Collecting information about your inventory
Processing inventory records

Username: harpew
Gateway : 10.241.80.xxx

Machines:

 Name       | IPs                       | Connectable   | Type      | Version   | Powered   | Console
------------+---------------------------+---------------+-----------+-----------+-----------+------------------------------------------------------------------
 BH10       | 192.168.1.164             | False         | Windows   | 10        | On        | 
 BH8007-1   | 192.168.1.71              | True          | OneFS     | 8.0.0.7   | On        | 
 BH8007-2   | 192.168.1.72              | True          | OneFS     | 8.0.0.7   | On        | 
 BH8007-3   | 192.168.1.73              | True          | OneFS     | 8.0.0.7   | On        | 
 BH8120-1   | 192.168.1.21              | False         | OneFS     | 8.1.2.0   | On        | 
 BH8120-2   | 192.168.1.22              | False         | OneFS     | 8.1.2.0   | On        | 
 BH8120-3   | 192.168.1.23              | False         | OneFS     | 8.1.2.0   | On        | 
 harpew2012 | fe80::8808:5fc8:250d:b93c | True          | WinServer | 2012R2    | On        | 
                      | 192.168.1.163             |               |           |           |           |

**Comment by willnx** [comment]
@bharpe Please do not comment on closed issues; it's like adding a comment to closed SR/bug/general ticket. I filed https://github.com/willnx/vlab/issues/24 for you. If you have any questions about the `connectable` column after checking out https://github.com/willnx/vlab/issues/24, please add a comment to that issue.

Thanks!
